music21/romanText/tsvConverter.py

```diff
@@ -166,81 +305,149 @@ def tabToM21(self):
+            ):
+                # if self.local_key contains a roman numeral, express it
+                # as a pitch, relative to the global key
+                localKeyNonRoman = getLocalKey(self.local_key, self.global_key)
+            else:
+                # otherwise, we assume self.local_key is already a pitch and
+                # pass it through unchanged
+                localKeyNonRoman = self.local_key
             thisEntry = roman.RomanNumeral(combined, localKeyNonRoman)
-            thisEntry.quarterLength = self.length
 
-            thisEntry.op = self.op
-            thisEntry.no = self.no
-            thisEntry.mov = self.mov
+            if self.dcml_version == 1:
+                thisEntry.quarterLength = self.length
+                # following metadata attributes seem to be missing from
+                # dcml_version 2 tsv files
+                thisEntry.op = self.op
+                thisEntry.no = self.no
+                thisEntry.mov = self.mov
 
             thisEntry.pedal = self.pedal
 
             thisEntry.phraseend = None
 
-        else:  # handling case of '@none'
-            thisEntry = note.Rest()
-            thisEntry.quarterLength = self.length
-
         return thisEntry
 
-# ------------------------------------------------------------------------------
+class TabChord(TabChordBase):
+    '''
+    An intermediate representation format for moving between tabular data in
+    DCML v1 and music21 chords.
+    '''
+    _dcml_version = 1
+    def __init__(self):
+        # self.numeral and self.relativeroot defined in super().__init__()
+        super().__init__()
+        self.chord = None
+        self.altchord = None
+        self.measure = None
+        self.beat = None
+        self.totbeat = None
+        self.timesig = None
+        self.op = None
+        self.no = None
+        self.mov = None
+        self.length = None
+        self.global_key = None
+        self.local_key = None
+        self.pedal = None
+        self.form = None
+        self.figbass = None
+        self.changes = None
+        self.phraseend = None
+
+    @property
+    def dcml_version(self):
+        return self._dcml_version
+
 
 
-def makeTabChord(row):
+class TabChordV2(TabChordBase):
     '''
-    Makes a TabChord out of a list imported from TSV data
-    (a row of the original tabular format -- see TsvHandler.importTsv()).
+    An intermediate representation format for moving between tabular data in
+    DCML v2 and music21 chords.
+    '''
+    _dcml_version = 2
+    def __init__(self):
+        # self.numeral and self.relativeroot defined in super().__init__()
+        super().__init__()
+        self.chord = None
+        self.mn = None
+        self.mn_onset = None
+        self.timesig = None
+        self.globalkey = None
+        self.localkey = None
+        self.pedal = None
+        self.form = None
+        self.figbass = None
+        self.changes = None
+        self.phraseend = None
```

Comment:
These attributes correspond to the headers of the columns of the DCML tsv files. The rationale for defining them separately in each subclass is that it makes it easier to compare the columns since they are all in one place. I can certainly change it if you prefer though.